CONTRIBUTING.md

```diff
@@ -60,8 +60,63 @@ make help
 
 ---
 
+## Local development using a container
 
-### Useful Links:
+If you'd like to skip all the bells and whistles of setting up the project, you can get up and running quickly using an
+"All-In-One" Docker container. It's a wrapper around Horust that enables you to compile it and all of its dependencies,
+and then work off the binary instead of compiling yourself.
+
+The `Makefile` contains a set of commands intended to scaffold an AIO container from scratch.
+ 
+Just run `make dargo-prep` inside the project's root folder.
+This will:
+
+1. Create a Docker image with a pre-determined working directory
+2. Run an (interactive / long-running) container off that image, with the local Horust project folder bind-mounted to the working directory
+3. Compile Horust and all its dependencies inside the container, using the local folder as storage for the cache
+4. Compile test dependencies and fill a few more caches (using `cargo test` and `cargo check`)
+
+When the Makefile target finishes, you will have a running container on your machine that you can compile Horust in.
+That container allows you to take advantage of `rustc`'s incremental compilation, without compiling locally.
+
+You can now run `make dargo COMMAND=X`, where `X` is some `cargo` command (like `build` / `test` / `check`) to run `cargo`
+inside that container with command `X`.
+
+If you like to go for maximum ergonomics, run the following command (swapping `~/.bashrc` for `~/.zshrc` or wherever you keep your shell stuff):
+ 
+```bash
+echo 'dargo(){ make dargo COMMAND=$1}' >> ~/.bashrc
+source ~/.bashrc
+```
+
+This will now enable you to run `dargo X` instead of `make dargo COMMAND=X`, to get a more `cargo`-like feel while using the container.
+Try running `dargo check` to see how if feels!
+
+### Container-based development example
+
+Just to make sure everything is clear, let's run a step-by-step mini tutorial on developing in container-based workflow "mode":
+
+1. Clone the repo to your local machine:
+```shell 
+git clone https://github.com/FedericoPonzi/Horust.git`
+cd Horust
+```
+2. Create the AIO container (can take a while):
+```shell
+make dargo-prep
+``` 
+3. Add "alias" to `make dargo COMMAND=X`:
```

Comment:
I would move it at the bottom as a suggested improvement for the workflow, it's not really mandatory required (and people not familiar with it or running another shell might get a different result)

Reply:
I just saw that you already wrote this on the previous paragraph. Then I think this step can be removed without arm, you might want to add a note at step 5 for the aliasing.

Reply:
Re-wrote the thing, take a look?

Reply:
Re-wrote the thing, take a look?